packages/experimental/Avatar/src/Avatar.styling.ts

```diff
@@ -0,0 +1,147 @@
+import { avatarName, AvatarTokens, AvatarSlotProps, AvatarProps } from './Avatar.types';
+import { Theme, UseStylingOptions, buildProps } from '@fluentui-react-native/framework';
+// import { borderStyles, fontStyles } from '@fluentui-react-native/tokens';
+// import { foregroundColorTokens } from '@fluentui-react-native/tokens';
+import { AvatarStates, defaultAvatarTokens } from './AvatarTokens';
+import { ViewStyle } from 'react-native';
+import { calculateEffectiveSizes, convertCoinColorFluent, getRingThickness } from './Avatar.helpers';
+
+const nameMap: { [key: string]: string } = {
+  start: 'flex-start',
+  center: 'center',
+  end: 'flex-end',
+};
+
+export const stylingSettings: UseStylingOptions<AvatarProps, AvatarSlotProps, AvatarTokens> = {
+  tokens: [defaultAvatarTokens, avatarName],
+  states: AvatarStates,
+  slotProps: {
+    root: buildProps(
+      (tokens: AvatarTokens) => {
+        const { physicalSize } = calculateEffectiveSizes(tokens);
+        const { horizontalIconAlignment, verticalIconAlignment } = tokens;
+        return {
+          style: {
+            flexDirection: 'row',
+            width: physicalSize,
+            height: physicalSize,
+            justifyContent: nameMap[horizontalIconAlignment || 'end'] as ViewStyle['justifyContent'],
+            alignItems: nameMap[verticalIconAlignment || 'end'] as ViewStyle['alignItems'],
+            horizontalIconAlignment,
+            verticalIconAlignment,
+            // color: tokens.coinColorFluent,
+            // iconStrokeColor: 'white',
+            // backgroundColor: ,
+          },
+        };
+      },
+      ['horizontalIconAlignment', 'verticalIconAlignment'],
+    ),
+    initials: buildProps((tokens: AvatarTokens) => {
+      const { initialsSize } = calculateEffectiveSizes(tokens);
+      return {
+        style: {
+          fontSize: initialsSize,
+        },
+      };
+    }, []),
```

Comment:
The cache key (the array) should include a list of tokens that affect the outcome out calculateEffectiveSizes

Reply:
Please also check the other buildProps functions and ensure the cache key is set up properly for each function.

Reply:
I added, thank you